packages/web/src/ui/views/viewer.js

```diff
@@ -83,47 +89,60 @@ const viewer = (state, i18n) => {
       .forEach((x) => {
         const updated = orbitControls.zoomToFit({ controls, camera, entities: prevEntities })
         controls = { ...controls, ...updated.controls }
+        moveRender()
       })
 
     const doRotatePanZoom = () => {
+      let changed = false
+
       if (rotateDelta[0] || rotateDelta[1]) {
         const updated = orbitControls.rotate({ controls, camera, speed: rotateSpeed }, rotateDelta)
         rotateDelta = [0, 0]
         controls = { ...controls, ...updated.controls }
+        changed = true
       }
 
       if (panDelta[0] || panDelta[1]) {
         const updated = orbitControls.pan({ controls, camera, speed: panSpeed }, panDelta)
         panDelta = [0, 0]
         camera.position = updated.camera.position
         camera.target = updated.camera.target
+        changed = true
       }
 
       if (zoomDelta) {
         const updated = orbitControls.zoom({ controls, camera, speed: zoomSpeed }, zoomDelta)
         controls = { ...controls, ...updated.controls }
         zoomDelta = 0
+        changed = true
       }
+      return changed
     }
 
     // the heart of rendering, as themes, controls, etc change
     const updateAndRender = (timestamp) => {
-      doRotatePanZoom()
+      if (doRotatePanZoom() || controls.autoRotate.enabled) {
+        moveRender()
+      }
 
-      const updated = orbitControls.update({ controls, camera })
-      controls = { ...controls, ...updated.controls }
-      camera.position = updated.camera.position
-      perspectiveCamera.update(camera)
+      if (!renderUntil || renderUntil > Date.now()) {
```

Comment:
Just one more really small request... sorry. Set...
`let renderUntil = Number.MAX_VALUE`
And then this IF only needs...
`if (renderUntil > Date.now())`

Reply:
np, it is a nice hint. Thnx!